couchbase/admin/query/v1/query.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+option go_package = "github.com/couchbase/goprotostellar/genproto/admin_query_v1;admin_query_v1";
+option ruby_package = "Couchbase::Protostellar::Generated::Admin::Query::V1";
+option php_namespace = "Couchbase\\Protostellar\\Generated\\Admin\\Query\\V1";
+option java_package = "com.couchbase.client.protostellar.admin.query.v1";
+option java_multiple_files = true;
+
+package couchbase.admin.query.v1;
+
+service QueryAdminService {
+  rpc GetAllIndexes(GetAllIndexesRequest) returns (GetAllIndexesResponse) {}
+
+  rpc CreatePrimaryIndex(CreatePrimaryIndexRequest)
+      returns (CreatePrimaryIndexResponse) {}
+  rpc CreateIndex(CreateIndexRequest) returns (CreateIndexResponse) {}
+
+  rpc DropPrimaryIndex(DropPrimaryIndexRequest)
+      returns (DropPrimaryIndexResponse) {}
+  rpc DropIndex(DropIndexRequest) returns (DropIndexResponse) {}
+
+  rpc BuildDeferredIndexes(BuildDeferredIndexesRequest)
+      returns (BuildDeferredIndexesResponse) {}
+}
+
+enum IndexType {
+  INDEX_TYPE_VIEW = 0;
+  INDEX_TYPE_GSI = 1;
+}
+
+enum IndexState {
+  INDEX_STATE_DEFERRED = 0;
+  INDEX_STATE_BUILDING = 1;
+  INDEX_STATE_PENDING = 2;
+  INDEX_STATE_ONLINE = 3;
+  INDEX_STATE_OFFLINE = 4;
+  INDEX_STATE_ABRIDGED = 5;
+  INDEX_STATE_SCHEDULED = 6;
+}
+
+message GetAllIndexesRequest {
+  optional string bucket_name = 1;
+  optional string scope_name = 2;
+  optional string collection_name = 3;
+}
+
+message GetAllIndexesResponse {
+  message Index {
+    string bucket_name = 1;
+    string scope_name = 2;
+    string collection_name = 3;
+    string name = 4;
+    bool is_primary = 5;
+    IndexType type = 6;
+    IndexState state = 7;
+    repeated string fields = 8;
+    optional string condition = 9;
+    optional string partition = 10;
+  }
+
+  repeated Index indexes = 1;
+}
+
+message CreatePrimaryIndexRequest {
+  string bucket_name = 1;
+  optional string scope_name = 2;
+  optional string collection_name = 3;
+  optional string name = 4;
+  optional int32 num_replicas = 5;
+  optional bool deferred = 6;
+}
```

Comment:
That was my original idea, though I did notice later that there is an 'IGNORE IF EXISTS' keyword setup for the query itself, so it may make sense to expose it as an actual field in Protostellar.  What are your thoughts?

Reply:
Filed an issue to track and discuss this outside the scope of this PR: https://couchbasecloud.atlassian.net/browse/ING-518